src/selector-native.js

```diff
@@ -62,18 +74,59 @@ jQuery.extend( {
 				}
 			}
 		} else {
-			jQuery.merge( results, context.querySelectorAll( selector ) );
+
+			newSelector = selector;
+			newContext = context;
+
+			// qSA considers elements outside a scoping root when evaluating child or
+			// descendant combinators, which is not what we want.
+			// In such cases, we work around the behavior by prefixing every selector in the
+			// list with an ID selector referencing the scope context.
+			// The technique has to be used as well when a leading combinator is used
+			// as such selectors are not recognized by querySelectorAll.
+			// Thanks to Andrew Dupont for this technique.
+			if ( nodeType === 1 &&
+				( rdescend.test( selector ) || rcombinators.test( selector ) ) ) {
```

Comment:
`rcombinators` is a much less clear name than e.g. `rleadingCombinator`, but I'm weighing the costs of gratuitous divergence from Sizzle. I think it would be best to rename in both.

Reply:
Makes sense to me.

Reply:
...with Sizzle going away soon, though, perhaps it'd be fine to just rename it on `main` & `3.x-stable`?

Reply:
Renamed here; I can submit a separate PR just renaming this on `3.x-stable`.

Reply:
Oh, well, this is a simple change so I submitted a PR to Sizzle as well:
* jQuery `3.x-stable`: #5208
* Sizzle `main`: https://github.com/jquery/sizzle/pull/494